core/benchmark/block_execution_benchmark.cpp

```diff
@@ -0,0 +1,304 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "benchmark/block_execution_benchmark.hpp"
+
+#include <algorithm>
+#include <numeric>
+
+#include "blockchain/block_tree.hpp"
+#include "primitives/runtime_dispatch_info.hpp"
+#include "runtime/runtime_api/core.hpp"
+#include "storage/trie/trie_storage.hpp"
+
+OUTCOME_CPP_DEFINE_CATEGORY(kagome::benchmark,
+                            BlockExecutionBenchmark::Error,
+                            e) {
+  switch (e) {
+    using E = kagome::benchmark::BlockExecutionBenchmark::Error;
+    case E::BLOCK_WEIGHT_DECODE_FAILED:
+      return "Failed to decode block weight";
+  }
+  return "Unknown BlockExecutionBenchmark error";
+}
+
+#define STR_SIMPLE(a) #a
+#define STR(a) STR_SIMPLE(a)
+#define CONCAT_SIMPLE(a, b) a##b
+#define CONCAT(a, b) CONCAT_SIMPLE(a, b)
+
+static_assert(STR(CONCAT(2, 3)) == std::string_view("23"));
+
+#define OUTCOME_TRY_MSG(var, expr, msg, ...)              \
+  auto CONCAT(_res_, __LINE__) = (expr);                  \
```

Comment:
boost uses `__COUNTER__` (`__LINE__` may cause duplicate/conflict) like `#define FOO_IMPL(name); #define FOO() FOO_IMPL(CONCAT(..., __COUNTER__));`

Reply:
Eh, it's local to this .cpp file so I wouldn't bother. I will leave a comment in case someone wants to move it to global scope.